grype/pkg/csv_provider.go

```diff
@@ -0,0 +1,132 @@
+package pkg
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/facebookincubator/nvdtools/wfn"
+	"github.com/mitchellh/go-homedir"
+
+	"github.com/anchore/grype/internal/log"
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/syft/pkg"
+)
+
+type errEmptyCSV struct {
+	csvFilepath string
+}
+
+func (e errEmptyCSV) Error() string {
+	return fmt.Sprintf("CSV file is empty: %s", e.csvFilepath)
+}
+
+func csvProvider(userInput string) ([]Package, Context, error) {
+	p, err := getCSVPackages(userInput)
+	return p, Context{}, err
+}
+
+func getCSVPackages(userInput string) ([]Package, error) {
+	reader, err := getCSVReader(userInput)
+	if err != nil {
+		return nil, err
+	}
+
+	by, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read csv: %w", err)
+	}
+
+	return decodeCSV(bytes.NewReader(by))
+}
+
+func decodeCSV(reader io.Reader) ([]Package, error) {
+	csvReader := csv.NewReader(reader)
+	records, err := csvReader.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse csv: %w", err)
+	}
+
+	packages := []Package{}
+
+	for _, row := range records {
+		rawCpe := strings.TrimSpace(row[0])
```

Comment:
question: do we need CPEs? That is, could we accept a list of only pURLs?

Reply:
We definitely don't need a cpe, as we can generate a cpe set using the information from a purl. Would you rather allow a cpe as optional or only allow purl?